src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue

```diff
@@ -632,8 +741,56 @@ export default {
 				&& this.message.id === this.message.threadId
 		},
 
+		isModerator() {
+			return this.$store.getters.isModerator
+		},
+
 		isModeratorOrOwner() {
-			return this.isCurrentUserOwnMessage || this.$store.getters.isModerator
+			return this.isCurrentUserOwnMessage || this.isModerator
+		},
+
+		isMessagePinned() {
+			if (!this.sharedItemsStore.sharedItems(this.message.token)) {
+				return false
+			}
+			return this.sharedItemsStore.sharedItems(this.message.token).pinned
+				&& Object.keys(this.sharedItemsStore.sharedItems(this.message.token).pinned).includes(this.message.id.toString())
+		},
+
+		pinInfo() {
+			if (!this.isMessagePinned) {
+				return null
+			}
+			return this.sharedItemsStore.sharedItems(this.message.token).pinned[this.message.id.toString()].metaData
+		},
```

Comment:
Its' in the chat message actions, so it's in the message store already:

```suggestion
		pinInfo() {
			return this.message.metaData
		},

		isMessagePinned() {
				return !!this.pinInfo?.pinnedAt
		},
```